modules/ballerina-core/src/main/java/org/ballerinalang/model/util/JSONUtils.java

```diff
@@ -661,7 +662,14 @@ private static boolean jsonNodeToBool(JsonNode jsonNode) {
         Iterator<Entry<String, JsonNode>> fields = jsonNode.fields();
         while (fields.hasNext()) {
             Entry<String, JsonNode> field = fields.next();
-            map.put(field.getKey(), getBValue(field.getValue()));
+            BValue bValue = getBValue(field.getValue());
+            if (mapType.getConstrainedType() == BTypes.typeAny ||
+                    mapType.getConstrainedType().equals(bValue.getType())) {
+                map.put(field.getKey(), bValue);
+            } else {
+                throw BLangExceptionHelper.getRuntimeException(RuntimeErrors.INCOMPATIBLE_TYPE_FOR_CASTING_JSON,
```

Comment:
This is not the correct way of handling ballerina runtime error. It seems like we have used this everywhere in this class. Please create an issue to track this.

Reply:
+1, for consistency I ll keep this error since it is the same error used for all the cases. I will create a issue for this.